api/tasks.go

```diff
@@ -62,3 +62,43 @@ func (api *JobStoreAPI) CreateTaskHandler(ctx context.Context) http.HandlerFunc
 		}
 	}
 }
+
+// GetTaskHandler returns a function that gets a specific task, associated with an existing Job resource, using the job id and task name passed in.
+func (api *JobStoreAPI) GetTaskHandler(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		vars := mux.Vars(req)
+		id := vars["id"]
+		taskName := vars["task_name"]
+		logData := log.Data{"job_id": id, "task_name": taskName}
+
+		task, err := api.jobStore.GetTask(req.Context(), id, taskName)
+		if err != nil {
+			log.Error(ctx, "getting task failed", err, logData)
+			if err == mongo.ErrJobNotFound {
+				http.Error(w, "Failed to find job that has the specified id", http.StatusNotFound)
```

Comment:
Thanks for explaining - I've changed them to lower case now. But I've kept this message as "failed to find task for the specified job id" instead of "failed to find task, job id not recognised" because the message will occur if the job is recognised but the task does not exist for that job.

Reply:
Not for this error - mongo.ErrJobNotFound - the job doesn't exist and hence the task doesn't exist, please check suggested changes.

Be wary of comparing line numbers here to that of your other branch once you've made a change. The line numbers will go out of sync and I think this has led you to this confusion of which error message Im referring too.

Basically whether the job id or task name is incorrect the action is to get a task, hence why I suggested changing the phrase to be about the task.

Think it from the perspective of the API user/service they are trying to retrieve a task not job